src/SchemaGenerator.ts

```diff
@@ -69,7 +69,8 @@ export class SchemaGenerator {
         const children = this
             .typeFormatter.getChildren(rootType)
             .filter((child) => child instanceof DefinitionType)
-            .filter((child: DefinitionType) => {
+            .map(child => child as DefinitionType)
```

Comment:
I like the base idea of this change but I don't like the implementation. You are adding real JavaScript code to the chain just for typing purposes. So after compiling to JavaScript it is a useless no-brainer (`child => child`). I think the best solution would be to convert the first filter function into a type guard:

```ts
.filter((child): child is DefinitionType => child instanceof DefinitionType)
.filter((child) => {
   ...
})
```

And by the way: The ` as DefinitionType[]` at the end of the chain is also no longer needed with this change.

Reply:
> I think the best solution would be to convert the first filter function into a type guard:
> ...

Agreed, I've just adapted the PR accordingly. Thanks for pointing this out!